src/plugins/plugin.tooltip.js

```diff
@@ -505,9 +522,10 @@ export class Tooltip extends Element {
 
   update(changed) {
     const me = this;
-    const options = me.options;
+    const options = me.options.setContext(me.getContext());
```

Comment:
I'm not sure, but could the `me.options` object just have been replaced here? (and probably only if `changed` is true)

Reply:
Yup, I could do that which would make a lot of the code simpler

Reply:
Thought of a reason not to do this. When we call set context, the old proxy is wrapped in a new proxy. That leads to a memory leak if we store it and continuously update.